### PR TITLE
Add support for Permissions Boundary on Function

### PR DESCRIPTION
**Pull request by jasonmk**
*Issue #, if available:*
#661 
*Description of changes:*
Adds support for specifying permissions boundary on a serverless function. Includes test templates

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jasonmk** [comment]
Updated with globals support.  Cheers!

File: requirements/dev.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@ coverage>=4.4.0
 flake8>=3.3.0
 tox>=2.2.1
 pytest-cov>=2.4.0
-pylint>=1.7.2
+pylint>=1.7.2,<2.0
```
Comment (keetonian) [inline requirements/dev.txt]: Any particular reason why this was necessary?

Comment (jasonmk) [reply requirements/dev.txt]: pip was trying to install a post 2.0 version which required py3.

Comment (keetonian) [reply requirements/dev.txt]: Ahh ok, thanks

**Pull request merged**